hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -1126,19 +1127,22 @@ def draw(
         if self.depth >= MAX_DEPTH:
             self.mark_invalid("max depth exceeded")
 
+        # Jump directly to the unwrapped strategy for the label and for do_draw.
+        # This avoids adding an extra span to all lazy strategies.
+        unwrapped = unwrap_strategies(strategy)
         if label is None:
-            assert isinstance(strategy.label, int)
-            label = strategy.label
+            label = unwrapped.label
+            assert isinstance(label, int)
+
         self.start_span(label=label)
         try:
             if not at_top_level:
-                return strategy.do_draw(self)
+                return unwrapped.do_draw(self)
             assert start_time is not None
             key = observe_as or f"generate:unlabeled_{len(self.draw_times)}"
             try:
-                strategy.validate()
```

Comment:
I'm pretty surprised about this, but this seems to duplicate another `strategy.validate()` we have above? Dropping this will probably result in a nice speedup..